lib/Service/YAPServer.hpp

```diff
@@ -0,0 +1,632 @@
+
+            if(true == send(CONTROL_CHANNEL_NUMBER, buf, sizeof(buf)))
+            {
+                /* Save Name and Callback for channel creation after response */
+                memcpy(m_pendingSuscribeChannel.m_name, &buf[CONTROL_CHANNEL_PAYLOAD_INDEX], CHANNEL_NAME_MAX_LEN);
+                m_pendingSuscribeChannel.m_callback = callback;
+            }
+        }
+    }
+
+    /**
+     * Returns current Sync state of th YAP Server.
+     */
+    bool isSynced()
+    {
+        return m_isSynced;
+    }
+
+    /**
+     * Get the number of configured channels.
+     * @returns Number of configured Data Channels. The Command Channel is ignored here.
+     */
+    uint8_t getNumberOfChannels()
+    {
+        uint8_t count = 0U;
+
+        for (uint8_t idx = 0; idx < tMaxChannels; idx++)
+        {
+            if (nullptr != m_dataChannels[idx].m_callback)
+            {
+                count++;
+            }
+        }
+
+        return count;
+    }
+
+private:
+    /**
+     * Control Channel Command: SYNC
+     * @param[in] payload Incomming Command data
+     */
+    void cmdSYNC(const uint8_t* payload)
+    {
+        uint8_t buf[CONTROL_CHANNEL_PAYLOAD_LENGTH] = {COMMANDS::SYNC_RSP, payload[0U], payload[1U], payload[2U],
+                                                       payload[3U]};
+        /* Ignore return as SYNC_RSP can fail */
+        (void) send(CONTROL_CHANNEL_NUMBER, buf, sizeof(buf));
+    }
+
+    /**
+     * Control Channel Command: SYNC_RSP
+     * @param[in] payload Incomming Command data
+     */
+    void cmdSYNC_RSP(const uint8_t* payload)
+    {
+        uint32_t rcvTimestamp = 0;
+
+        /* Using (payloadSize - 1U) as CMD Byte is not passed. */
+        if (true == Util::byteArrayToUint32(payload, sizeof(uint32_t), rcvTimestamp))
+        {
+            /* Check Timestamp with m_lastSyncCommand */
+            if (rcvTimestamp == m_lastSyncCommand)
+            {
+                m_lastSyncResponse = m_lastSyncCommand;
+                m_isSynced         = true;
+            }
+            else
+            {
+                m_isSynced = false;
+            }
+        }
+    }
+
+    /**
+     * Control Channel Command: SCRB
+     * @param[in] payload Incomming Command data
+     */
+    void cmdSCRB(const uint8_t* payload)
+    {
+        uint8_t channelNumber = getChannelNumber(reinterpret_cast<const char*>(payload));
+        uint8_t buf[CONTROL_CHANNEL_PAYLOAD_LENGTH] = {0U};
+        buf[0U] = COMMANDS::SCRB_RSP;
+
+        if (CONTROL_CHANNEL_NUMBER != channelNumber)
+        {
+            buf[1U] = channelNumber;
+            buf[2U] = getChannelDLC(channelNumber);
+        }
+        else
+        {
+            /* No channel found. Invalid Response: Channel 0 with DLC = 0 */
+            buf[1U] = 0U;
+            buf[2U] = 0U;
+        }
+
+        /* Ignore return as SCRB_RSP can fail. */
+        (void) send(CONTROL_CHANNEL_NUMBER, buf, sizeof(buf));

```

Comment:
Maybe implement some kind of retry in the future?

Reply:
Yes, or simply go out of sync in case it happens. The other side will go out of sync anyway.